authors/apps/articles/tests/api/test_tags.py

```diff
@@ -0,0 +1,262 @@
+import json
+import random
+import string
+
+from rest_framework import status
+from rest_framework.reverse import reverse
+
+from authors.apps.articles.tests.api.test_articles import BaseArticlesTestCase
+
+
+class BaseTagsTestCase(BaseArticlesTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.tags = {
+            "tags": [
+                "Andela",
+                "Cohort 30",
+                "MacBook Pro 2016"
+            ]
+        }
+        # create an article
+        self.article_slug = self.create_article(published=True)['slug']
+
+    def generate_url(self, slug):
+        """
+        Generate tagging url for an article
+        :param slug:
+        :return:
+        """
+        return reverse("articles:article-tags", kwargs={'slug': slug or self.article_slug})
+
+    def tag_article(self, slug=None, tags=None):
+        """
+        Helper method to tag an article
+        :param slug:
+        :param tags:
+        :return:
+        """
+        return self.client.post(self.generate_url(slug or self.article_slug), tags or self.tags, format="json")
```

Comment:
here you are repeating yourself by passing a `self.article_slug` to the `generate_url` method.